crates/ra_syntax/src/parsing/reparsing.rs

```diff
@@ -162,20 +162,27 @@ fn is_balanced(tokens: &[Token]) -> bool {
 fn merge_errors(
     old_errors: Vec<SyntaxError>,
     new_errors: Vec<SyntaxError>,
-    old_range: TextRange,
+    range_before_reparse: TextRange,
     edit: &AtomTextEdit,
 ) -> Vec<SyntaxError> {
     let mut res = Vec::new();
-    for e in old_errors {
-        if e.offset() <= old_range.start() {
```

Comment:
**AHTUNG!** I suppose in this if statement we had a mistake, we need to check that `e.range().end() <= old_range.start()` and I corrected it in this PR (tests pass anyway...)

Reply:
Nit: It's a**c**htung :) Will take a look at the PR sometime soon